plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -0,0 +1,293 @@
+
+    if (!encodedClusters) {
+      return {};
+    }
+
+    try {
+      const decodedClusters = Buffer.from(encodedClusters, 'base64').toString();
+      const clusters: KubernetesProxyClusters = JSON.parse(decodedClusters);
+      return clusters;
+    } catch (e: any) {
+      this.logger.debug(
+        `error with encoded cluster header: ${stringifyError(e)}`,
+      );
+    }
+    return {};
+  }
+
+  private async getClusterDetails(
+    clusterSupplier: KubernetesClustersSupplier,
+    krc: KubernetesProxyClusters,
+  ): Promise<ClusterDetails[]> {
+    const clusters = await clusterSupplier.getClusters();
+
+    const clusterNames = Object.keys(krc);
+
+    const clusterDetails = clusters.filter(c => clusterNames.includes(c.name));
+
+    const clusterDetailsAuth = clusterDetails.map(c => {
+      const cAuth: ClusterDetails = Object.assign(c, {
+        serviceAccountToken: krc[c.name],
+      });
+      return cAuth;
+    });
+
+    return clusterDetailsAuth;
+  }
+
+  private getClusterURI(details: ClusterDetails): string {
+    const client = this.getKubeConfig(details);
+    return client.getCurrentCluster()?.server || '';
+  }
+
+  private async makeRequestToCluster(
+    details: ClusterDetails,
+    req: Request,
+  ): Promise<KubernetesProxyResponse> {
+    const serverIP = this.getClusterURI(details);
+
+    if (!serverIP) {
+      this.logger.error(`Cluster ${details.name} details IP error`);
+
+      throw new ConflictError('Cluster detail error');
+    }
+
+    const query = decodeURIComponent(req.params.encodedQuery) || '';
+    const uri = `${serverIP}/${query}`;
+
+    const contentType = req.header(HEADER_CONTENT_TYPE) || APPLICATION_JSON;
+
+    return await this.sendClusterRequest(
+      details,
+      uri,
+      req.method,
+      contentType,
+      req.body,
+    );
+  }
+
+  private async sendClusterRequest(
+    details: ClusterDetails,
+    uri: string,
+    method: string,
+    contentType: string,
+    body?: any,
+  ): Promise<KubernetesProxyResponse> {
+    const bearerToken = details.serviceAccountToken;
+
+    if (!bearerToken) {
+      this.logger.error('Invalid service account token');
+
+      throw new AuthenticationError('Invalid service account token');
+    }
+
+    const reqData: RequestInit = {
+      method,
+      headers: {
+        'Content-Type': contentType,
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    };
+
+    if (!details.skipTLSVerify) {
+      if (details.caData) {
+        const ca = bufferFromFileOrString('', details.caData)?.toString() || '';
+        reqData.agent = new https.Agent({ ca });
+      } else {
+        this.logger.error('could not find CA certificate!');
+        throw new InputError(
+          'Invalid CA certificate configured within Backstage',
+        );
+      }
+    }
+
+    if (body && Object.keys(body).length > 0) {
+      reqData.body = JSON.stringify(body);
+    }
+
+    try {
+      const res = await fetch(uri, reqData);
```

Comment:
@freben How do I deal with this? the URI comes from config or from the catalog as long as it matches with the requested cluster in the request's header, I think it will keep being detected as a `user-provided value`, should we ignore this line with a comment?

Reply:
I don't think codeql has suppression comments at all. We are expected to mark it as clean in here in the PR instead, so us reviewers will be responsible for dealing with it